qutebrowser/config/configcommands.py

```diff
@@ -250,6 +250,72 @@ def config_unset(self, option, temp=False):
         with self._handle_config_error():
             self._config.unset(option, save_yaml=not temp)
 
+    @cmdutils.register(instance='config-commands')
+    @cmdutils.argument('option', completion=configmodel.structure_option)
+    def config_add(
+            self, option, key=None, value=None, temp=False, replace=False
+    ):
+        """Adds an option.
+
+        This adds an element to a dictionary or list. --replace is needed
+        to override existing values.
+
+        Args:
+            option: The name of the option.
+            value: The value to place in either the list or the dictionary.
+            key: The key to set if it's a dictionary.
+            temp: Don't touch autoconfig.yml.
+            replace: Whether or not we should replace, default is not.
+        """
+        with self._handle_config_error():
+            option_value = self._config.get_obj(option)
+
+            # Attempt to replace if it's there.
+            if isinstance(option_value, list):
+                if value is None:
+                    # In this case our key is teh "value", since we are just
+                    # adding to the list.
+                    option_value.append(key)
+
+                if value and replace:
+                    # In this case we are trying to replace something at an
+                    # index.
```

Comment:
I'm not sure this is useful. At that point it's probably easier for the user to just use `:config-set` with the completion to get the existing value, and replace it there.